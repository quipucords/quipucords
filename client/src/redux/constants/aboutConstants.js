const ABOUT_DIALOG_OPEN = 'ABOUT_DIALOG_OPEN';
const ABOUT_DIALOG_CLOSE = 'ABOUT_DIALOG_CLOSE';

export { ABOUT_DIALOG_OPEN, ABOUT_DIALOG_CLOSE };
