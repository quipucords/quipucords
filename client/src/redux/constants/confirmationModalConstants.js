const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

export { CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
