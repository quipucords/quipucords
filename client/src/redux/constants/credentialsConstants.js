
const LOAD_CREDENTIALS_SUCCESS = 'LOAD_CREDENTIALS_SUCCESS';

export {
  LOAD_CREDENTIALS_SUCCESS
}
