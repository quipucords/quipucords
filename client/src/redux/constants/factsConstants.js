const ADD_FACTS_ERROR = 'ADD_FACTS_ERROR';
const ADD_FACTS_LOADING = 'ADD_FACTS_LOADING';
const ADD_FACTS_SUCCESS = 'ADD_FACTS_SUCCESS';

export { ADD_FACTS_ERROR, ADD_FACTS_LOADING, ADD_FACTS_SUCCESS };
