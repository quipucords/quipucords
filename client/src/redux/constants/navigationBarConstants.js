const NAV_TOGGLE_COLLAPSE = 'NAV_TOGGLE_COLLAPSE';
const NAV_COLLAPSE = 'NAV_COLLAPSE';
const NAV_OPEN = 'NAV_OPEN';

export { NAV_TOGGLE_COLLAPSE, NAV_COLLAPSE, NAV_OPEN };
