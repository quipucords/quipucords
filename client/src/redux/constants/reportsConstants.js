const GET_REPORT = 'GET_REPORT';
const GET_MERGE_REPORT = 'GET_MERGE_REPORT';

export { GET_REPORT, GET_MERGE_REPORT };
