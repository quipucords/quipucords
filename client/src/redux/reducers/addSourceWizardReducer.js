import { credentialsTypes, sourcesTypes } from '../constants';
import helpers from '../../common/helpers';
import _ from 'lodash';

const initialState = {
  view: {
    show: false,
    add: false,
    edit: false,
    allCredentials: [],
    source: {},
    error: false,
    errorMessage: null,
    stepOneValid: false,
    stepTwoValid: false,
    fulfilled: false
  }
};

function addSourceWizardReducer(state = initialState, action) {
  switch (action.type) {
    // Show/Hide
    case sourcesTypes.CREATE_SOURCE_SHOW:
      return helpers.setStateProp(
        'view',
        {
          show: true,
          add: true,
          allCredentials: state.view.allCredentials
        },
        {
          state,
          initialState
        }
      );

    case sourcesTypes.EDIT_SOURCE_SHOW:
      return helpers.setStateProp(
        'view',
        {
          show: true,
          edit: true,
          source: action.source,
          stepOneValid: true
        },
        {
          state,
          initialState
        }
      );

    case sourcesTypes.UPDATE_SOURCE_HIDE:
      return helpers.setStateProp(
        'view',
        {
          show: false
        },
        {
          state,
          initialState
        }
      );

    case sourcesTypes.UPDATE_SOURCE_WIZARD_STEPONE:
      return helpers.setStateProp(
        'view',
        {
          source: action.source,
          stepOneValid: true
        },
        {
          state,
          reset: false
        }
      );

    case sourcesTypes.UPDATE_SOURCE_WIZARD_STEPTWO:
      return helpers.setStateProp(
        'view',
        {
          source: action.source,
          stepTwoValid: true
        },
        {
          state,
          reset: false
        }
      );

    case sourcesTypes.INVALID_SOURCE_WIZARD_STEPTWO:
      return helpers.setStateProp(
        'view',
        {
          stepTwoValid: false
        },
        {
          state,
          reset: false
        }
      );

    // Error/Rejected
    case helpers.REJECTED_ACTION(sourcesTypes.ADD_SOURCE):
      return helpers.setStateProp(
        'view',
        {
          error: action.error,
          errorMessage: helpers.getErrorMessageFromResults(action.payload)
        },
        {
          state,
          reset: false
        }
      );

    case helpers.REJECTED_ACTION(sourcesTypes.UPDATE_SOURCE):
      return helpers.setStateProp(
        'view',
        {
          error: action.error,
          errorMessage: helpers.getErrorMessageFromResults(action.payload)
        },
        {
          state,
          reset: false
        }
      );

    // Success/Fulfilled
    case helpers.FULFILLED_ACTION(sourcesTypes.UPDATE_SOURCE):
    case helpers.FULFILLED_ACTION(sourcesTypes.ADD_SOURCE):
      return helpers.setStateProp(
        'view',
        {
          source: action.payload.data,
          fulfilled: true
        },
        {
          state,
          reset: false
        }
      );

    case helpers.FULFILLED_ACTION(credentialsTypes.GET_WIZARD_CREDENTIALS):
      return helpers.setStateProp(
        'view',
        {
          allCredentials: _.get(action, 'payload.data.results', [])
        },
        {
          state,
          reset: false
        }
      );

    default:
      return state;
  }
}

addSourceWizardReducer.initialState = initialState;

export { initialState, addSourceWizardReducer };

export default addSourceWizardReducer;
