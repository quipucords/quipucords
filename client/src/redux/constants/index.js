import * as aboutTypes from './aboutConstants';
import * as credentialsTypes from './credentialsConstants';
import * as factsTypes from './factsConstants';
import * as navigationBarTypes from './navigationBarConstants';
import * as reportsTypes from './reportsConstants';
import * as scansTypes from './scansConstants';
import * as sourcesTypes from './sourcesConstants';
import * as viewToolbarTypes from './viewToolbarConstants';

const reduxTypes = {
  about: aboutTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  navigation: navigationBarTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes,
  aboutTypes,
  credentialsTypes,
  factsTypes,
  navigationBarTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  viewToolbarTypes
};

export default reduxTypes;
