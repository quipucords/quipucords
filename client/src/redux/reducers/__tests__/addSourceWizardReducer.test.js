import addSourceWizardReducer from '../addSourceWizardReducer';
import { sourcesTypes as types, credentialsTypes } from '../../constants';
import helpers from '../../../common/helpers';

describe('AddSourceWizardReducer', () => {
  it('should return the initial state', () => {
    expect(addSourceWizardReducer.initialState).toBeDefined();
  });

  it('should handle specific defined types', () => {
    const specificTypes = [
      types.CREATE_SOURCE_SHOW,
      types.EDIT_SOURCE_SHOW,
      types.UPDATE_SOURCE_HIDE,
      types.UPDATE_SOURCE_WIZARD_STEPONE,
      types.UPDATE_SOURCE_WIZARD_STEPTWO,
      types.INVALID_SOURCE_WIZARD_STEPTWO
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: value
      };

      const resultState = addSourceWizardReducer(undefined, dispatched);

      expect({ type: value, result: resultState }).toMatchSnapshot(`defined type ${value}`);
    });
  });

  it('should handle all defined error types', () => {
    const specificTypes = [types.ADD_SOURCE, types.UPDATE_SOURCE];

    specificTypes.forEach(value => {
      const dispatched = {
        type: helpers.REJECTED_ACTION(value),
        error: true,
        payload: {
          message: 'MESSAGE',
          response: {
            status: 0,
            statusText: 'ERROR TEST',
            data: {
              detail: 'ERROR'
            }
          }
        }
      };

      const resultState = addSourceWizardReducer(undefined, dispatched);

      expect({ type: helpers.REJECTED_ACTION(value), result: resultState }).toMatchSnapshot(`rejected types ${value}`);
    });
  });

  it('should handle all defined fulfilled types', () => {
    const specificTypes = [types.ADD_SOURCE, types.UPDATE_SOURCE, credentialsTypes.GET_WIZARD_CREDENTIALS];

    specificTypes.forEach(value => {
      const dispatched = {
        type: helpers.FULFILLED_ACTION(value),
        payload: {
          data: {
            test: 'success'
          }
        }
      };

      const resultState = addSourceWizardReducer(undefined, dispatched);

      expect({ type: helpers.FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot(
        `fulfilled types ${value}`
      );
    });
  });
});
