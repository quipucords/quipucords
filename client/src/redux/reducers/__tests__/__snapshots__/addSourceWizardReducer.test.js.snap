// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defined type CREATE_SOURCE_SHOW 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": true,
      "allCredentials": Array [],
      "edit": false,
      "error": false,
      "errorMessage": null,
      "fulfilled": false,
      "show": true,
      "source": Object {},
      "stepOneValid": false,
      "stepTwoValid": false,
    },
  },
  "type": "CREATE_SOURCE_SHOW",
}
`;

exports[`defined type EDIT_SOURCE_SHOW 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": false,
      "allCredentials": Array [],
      "edit": true,
      "error": false,
      "errorMessage": null,
      "fulfilled": false,
      "show": true,
      "source": undefined,
      "stepOneValid": true,
      "stepTwoValid": false,
    },
  },
  "type": "EDIT_SOURCE_SHOW",
}
`;

exports[`defined type INVALID_SOURCE_WIZARD_STEPTWO 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": false,
      "allCredentials": Array [],
      "edit": false,
      "error": false,
      "errorMessage": null,
      "fulfilled": false,
      "show": false,
      "source": Object {},
      "stepOneValid": false,
      "stepTwoValid": false,
    },
  },
  "type": "INVALID_SOURCE_WIZARD_STEPTWO",
}
`;

exports[`defined type UPDATE_SOURCE_HIDE 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": false,
      "allCredentials": Array [],
      "edit": false,
      "error": false,
      "errorMessage": null,
      "fulfilled": false,
      "show": false,
      "source": Object {},
      "stepOneValid": false,
      "stepTwoValid": false,
    },
  },
  "type": "UPDATE_SOURCE_HIDE",
}
`;

exports[`defined type UPDATE_SOURCE_WIZARD_STEPONE 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": false,
      "allCredentials": Array [],
      "edit": false,
      "error": false,
      "errorMessage": null,
      "fulfilled": false,
      "show": false,
      "source": undefined,
      "stepOneValid": true,
      "stepTwoValid": false,
    },
  },
  "type": "UPDATE_SOURCE_WIZARD_STEPONE",
}
`;

exports[`defined type UPDATE_SOURCE_WIZARD_STEPTWO 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": false,
      "allCredentials": Array [],
      "edit": false,
      "error": false,
      "errorMessage": null,
      "fulfilled": false,
      "show": false,
      "source": undefined,
      "stepOneValid": false,
      "stepTwoValid": true,
    },
  },
  "type": "UPDATE_SOURCE_WIZARD_STEPTWO",
}
`;

exports[`fulfilled types ADD_SOURCE 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": false,
      "allCredentials": Array [],
      "edit": false,
      "error": false,
      "errorMessage": null,
      "fulfilled": true,
      "show": false,
      "source": Object {
        "test": "success",
      },
      "stepOneValid": false,
      "stepTwoValid": false,
    },
  },
  "type": "ADD_SOURCE_FULFILLED",
}
`;

exports[`fulfilled types GET_WIZARD_CREDENTIALS 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": false,
      "allCredentials": Array [],
      "edit": false,
      "error": false,
      "errorMessage": null,
      "fulfilled": false,
      "show": false,
      "source": Object {},
      "stepOneValid": false,
      "stepTwoValid": false,
    },
  },
  "type": "GET_WIZARD_CREDENTIALS_FULFILLED",
}
`;

exports[`fulfilled types UPDATE_SOURCE 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": false,
      "allCredentials": Array [],
      "edit": false,
      "error": false,
      "errorMessage": null,
      "fulfilled": true,
      "show": false,
      "source": Object {
        "test": "success",
      },
      "stepOneValid": false,
      "stepTwoValid": false,
    },
  },
  "type": "UPDATE_SOURCE_FULFILLED",
}
`;

exports[`rejected types ADD_SOURCE 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": false,
      "allCredentials": Array [],
      "edit": false,
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "show": false,
      "source": Object {},
      "stepOneValid": false,
      "stepTwoValid": false,
    },
  },
  "type": "ADD_SOURCE_REJECTED",
}
`;

exports[`rejected types UPDATE_SOURCE 1`] = `
Object {
  "result": Object {
    "view": Object {
      "add": false,
      "allCredentials": Array [],
      "edit": false,
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "show": false,
      "source": Object {},
      "stepOneValid": false,
      "stepTwoValid": false,
    },
  },
  "type": "UPDATE_SOURCE_REJECTED",
}
`;
